File: app/src/main/java/org/helpapaw/helpapaw/data/repositories/SettingsRepository.java
package org.helpapaw.helpapaw.data.repositories;

import android.content.SharedPreferences;

public class SettingsRepository implements ISettingsRepository {

    private SharedPreferences preferences;

    public SettingsRepository(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    @Override
    public void saveRadius(int radius) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt("radius", radius);
        editor.apply();
    }

    @Override
    public void saveTimeout(int timeout) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt("timeout", timeout);
        editor.apply();
    }

    @Override
    public int getRadius() {
        return preferences.getInt("radius", 10);
    }

    @Override
    public int getTimeout() {
        return preferences.getInt("timeout", 7);
    }

    @Override
    public double getLatitude() {
        String lat = preferences.getString("latitude", "0");
        return Double.valueOf(lat);
    }

    @Override
    public void setLatitude(double latitude) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString("latitude", String.valueOf(latitude));
        editor.apply();
    }

    @Override
    public double getLongitude() {
        String longi = preferences.getString("longitude", "0");
        return Double.valueOf(longi);
    }

    @Override
    public void setLongitude(double longitude) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString("longitude", String.valueOf(longitude));
        editor.apply();
    }

    @Override
    public float getZoom() {
        return preferences.getFloat("zoom", 0f);
    }

    @Override
    public void setZoom(float zoom) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putFloat("zoom", zoom);
        editor.apply();
    }
}

File: app/src/main/java/org/helpapaw/helpapaw/data/repositories/SettingsRepository.kt
package org.helpapaw.helpapaw.data.repositories

import android.content.SharedPreferences

class SettingsRepository(val preferences: SharedPreferences) : ISettingsRepository {


    companion object {
        const val RADIUS_FIELD = "signalRadius"
        const val TIMEOUT_FIELD = "signalTimeout"
        const val LAST_SHOWN_LATITUDE_FIELD = "lastShownLatitude"
        const val LAST_SHOWN_LONGITUDE_FIELD = "lastShownLongitude"
        const val LAST_SHOWN_ZOOM_FIELD = "lastShownZoom"

    }

    override fun saveRadius(radius: Int) {
        val editor = preferences.edit()
        editor.putInt(RADIUS_FIELD, radius)
        editor.apply()
    }

    override fun saveTimeout(timeout: Int) {
        val editor = preferences.edit()
        editor.putInt(TIMEOUT_FIELD, timeout)
        editor.apply()
    }

    override fun getRadius(): Int =
            preferences.getInt(RADIUS_FIELD, 10)


    override fun getTimeout(): Int =
            preferences.getInt(TIMEOUT_FIELD, 7)

    override fun getLastShownLatitude(): Double {
        val lat = preferences.getString(LAST_SHOWN_LATITUDE_FIELD, "0")
        return java.lang.Double.valueOf(lat!!)
    }

    override fun setLastShownLatitude(latitude: Double) {
        val editor = preferences.edit()
        editor.putString(LAST_SHOWN_LATITUDE_FIELD, latitude.toString())
        editor.apply()
    }

    override fun getLastShownLongitude(): Double {
        val longi = preferences.getString(LAST_SHOWN_LONGITUDE_FIELD, "0")
        return longi.toDouble()
    }

    override fun setLastShownLongitude(longitude: Double) {
        val editor = preferences.edit()
        editor.putString(LAST_SHOWN_LONGITUDE_FIELD, longitude.toString())
        editor.apply()
    }

    override fun getLastShownZoom(): Float =
            preferences.getFloat(LAST_SHOWN_ZOOM_FIELD, 0f)

    override fun setLastShownZoom(zoom: Float) {
        val editor = preferences.edit()
        editor.putFloat(LAST_SHOWN_ZOOM_FIELD, zoom)
        editor.apply()
    }

    override fun clearLocationData() {
        val editor = preferences.edit()
        editor.remove(LAST_SHOWN_LATITUDE_FIELD)
        editor.remove(LAST_SHOWN_LONGITUDE_FIELD)
        editor.remove(LAST_SHOWN_ZOOM_FIELD)
        editor.apply()
    }

}